pandas/core/sorting.py

```diff
@@ -417,11 +418,16 @@ def nargminmax(values, method: str):
     mask = np.asarray(isna(values))
     values = values._values_for_argsort()
 
-    idx = np.arange(len(values))
+    idx = np.arange(values.shape[axis])
+    if values.ndim > 1:
+        if mask.any():
```

Comment:
is this hit in tests?

Reply:
the case with mask.any() being True?  no

Reply:
updated to implement this, with tests that hit it